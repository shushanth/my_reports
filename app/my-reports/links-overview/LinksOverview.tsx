import { getLinksApi, Links } from "../../../mock_server/linksApi";
import StoreProvider from "../../StoreProvider";
import styles from "./linksOverview.module.scss";
import ReportsLinks from "./reports-links/ReportsLinks";

export default async function () {
  const linksResponse = (await getLinksApi()) as Links;
  const { items } = linksResponse;
  const linksData = items.map(({ id, domain, url, sourceType, updatedAt }) => {
    const date = new Date(updatedAt);
    const currentDate = date.toLocaleDateString();
    return {
      domain,
      url,
      sourceType,
      date: currentDate,
    };
  });
  return (
    <div className={styles.links_overview}>
      <StoreProvider>
        <ReportsLinks reports={linksData} />
      </StoreProvider>
    </div>
  );
}
