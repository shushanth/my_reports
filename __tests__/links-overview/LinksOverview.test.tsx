import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import LinksOverview from "@/app/my-reports/links-overview/LinksOverview";

describe("LinksOverview", () => {
  it("renders component", async () => {
    render(<LinksOverview />);
    const heading = screen.queryByTestId("linksOverview_page--component");
    expect(heading).toBeDefined();
  });

  it("renders component page unchanged", () => {
    const { container } = render(<LinksOverview />);
    expect(container).toMatchSnapshot();
  });
});
