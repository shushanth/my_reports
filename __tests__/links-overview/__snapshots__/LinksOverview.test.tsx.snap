// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinksOverview renders component page unchanged 1`] = `<div />`;
